mm2src/coins/utxo/utxo_common.rs

```diff
@@ -1460,44 +1456,63 @@ pub async fn sign_and_broadcast_taker_payment_spend<T: UtxoCommonOps>(
     Ok(final_tx)
 }
 
-pub fn send_taker_fee<T>(coin: T, fee_pub_key: &[u8], dex_fee: DexFee) -> TransactionFut
+pub fn send_taker_fee<T>(coin: T, dex_fee: DexFee) -> TransactionFut
 where
-    T: UtxoCommonOps + GetUtxoListOps,
+    T: UtxoCommonOps + GetUtxoListOps + SwapOps,
 {
-    let address = try_tx_fus!(address_from_raw_pubkey(
-        fee_pub_key,
+    let outputs = try_tx_fus!(generate_taker_fee_tx_outputs(&coin, &dex_fee,));
+
+    send_outputs_from_my_address(coin, outputs)
+}
+
+fn generate_taker_fee_tx_outputs<T>(coin: &T, dex_fee: &DexFee) -> Result<Vec<TransactionOutput>, String>
+where
+    T: UtxoCommonOps + SwapOps,
+{
+    let dex_address = address_from_raw_pubkey(
+        coin.dex_pubkey(),
         coin.as_ref().conf.address_prefixes.clone(),
         coin.as_ref().conf.checksum_type,
         coin.as_ref().conf.bech32_hrp.clone(),
         coin.addr_format().clone(),
-    ));
-
-    let outputs = try_tx_fus!(generate_taker_fee_tx_outputs(
-        coin.as_ref().decimals,
-        address.hash(),
-        &dex_fee,
-    ));
-
-    send_outputs_from_my_address(coin, outputs)
-}
+    )?;
+    let burn_address = address_from_raw_pubkey(
+        coin.burn_pubkey(),
+        coin.as_ref().conf.address_prefixes.clone(),
+        coin.as_ref().conf.checksum_type,
+        coin.as_ref().conf.bech32_hrp.clone(),
+        coin.addr_format().clone(),
+    )?;
 
-fn generate_taker_fee_tx_outputs(
-    decimals: u8,
-    address_hash: &AddressHashEnum,
-    dex_fee: &DexFee,
-) -> Result<Vec<TransactionOutput>, MmError<NumConversError>> {
-    let fee_amount = dex_fee.fee_uamount(decimals)?;
+    let fee_amount = dex_fee
+        .fee_amount_as_u64(coin.as_ref().decimals)
+        .map_err(|err| err.get_inner().to_string())?;
 
     let mut outputs = vec![TransactionOutput {
         value: fee_amount,
-        script_pubkey: Builder::build_p2pkh(address_hash).to_bytes(),
+        script_pubkey: Builder::build_p2pkh(dex_address.hash()).to_bytes(),
     }];
 
-    if let Some(burn_amount) = dex_fee.burn_uamount(decimals)? {
-        outputs.push(TransactionOutput {
-            value: burn_amount,
-            script_pubkey: Builder::default().push_opcode(Opcode::OP_RETURN).into_bytes(),
-        });
+    match dex_fee {
+        DexFee::Standard(_) => {}, // do nothing
```

Comment:
how about returning an error here since `DexFee::Standard` is not supported for utxo once this PR is merged?

Reply:
I guess, we cannot fail here for this time yet, because this function is also used for old (standard) dex fee validation at the maker side.
(added a comment to add an error when all nodes upgrade)